corehq/apps/userreports/forms.py

```diff
@@ -0,0 +1,33 @@
+from django import forms
+from django.utils.translation import ugettext as _
+from django.utils.translation import ugettext_lazy
+
+from crispy_forms import layout as crispy
+from crispy_forms.bootstrap import StrictButton
+from corehq.apps.hqwebapp.crispy import HQFormHelper
+
+from corehq.apps.hqwebapp import crispy as hqcrispy
+from corehq.apps.linked_domain.dbaccessors import (
+    get_linked_domains,
+)
+
+
+class CopyReportForm(forms.Form):
```

Comment:
The "copy" naming throughout this PR feels a bit off to me. It matches apps, but apps support basic copy functionality, that is, copying within the same domain without involving linked domains. Any idea if we'd consider supporting basic copy for reports? It seems not hard now that all of this work is done - not saying it should be part of this PR or your responsibility, just curious. Someone was asking about copying reports on field-dev the other day.

Reply:
I think this would totally work for copying reports, in the same way that apps are copied now. I think we'd need to think a bit about how to handle datasources if a report is copied. The answer might be "make a new datasource for each copy", but that seems wasteful if you are just making a new report to have a different default filter. 
I'll update the naming of everything here (you are right that I modeled this off apps, so didn't think too hard about this name).